gtda/diagrams/features.py

```diff
@@ -716,3 +716,184 @@ def transform(self, X, y=None):
+    PairwiseDistance
+
+    References
+    ----------
+    .. [1] M. Carrière, S. Y. Oudot, and M. Ovsjanikov, "Stable Topological
+           Signatures for Points on 3D Shapes"; *Computer Graphics Forum*
+           **34** (5), 2015; `DOI: 10.1111:cgf.12692`2015
+           <https://doi.org/10.1111:cgf.12692`2015>`_.
+
+    """
+
+    _hyperparameters = {
+        'n_distances': {'type': (int, type(None)),
+                        'in': Interval(1, np.inf, closed='left')},
+        'metric': {'type': (str, FunctionType)},
+        'metric_params': {'type': dict}
+        }
+
+    def __init__(self, n_distances=10, metric='chebyshev', metric_params={},
+                 n_jobs=None):
+        self.n_distances = n_distances
+        self.metric = metric
+        self.metric_params = metric_params
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y=None):
+        """Store all observed homology dimensions in
+        :attr:`homology_dimensions_` and compute :attr:`n_distances_` and
+        :attr:`effective_metric_params`. Then, return the estimator.
+
+        This method is here to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features, 3)
+            Input data. Array of persistence diagrams, each a collection of
+            triples [b, d, q] representing persistent topological features
+            through their birth (b), death (d) and homology dimension (q).
+            It is important that, for each possible homology dimension, the
+            number of triples for which q equals that homology dimension is
+            constants across the entries of `X`.
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        validate_params(
+            self.get_params(), self._hyperparameters, exclude=['n_jobs'])
+
+        # Find the unique homology dimensions in the 3D array X passed to `fit`
+        # assuming that they can all be found in its zero-th entry
+        homology_dimensions_fit = np.unique(X[0, :, 2])
+        self.homology_dimensions_ = \
+            _homology_dimensions_to_sorted_ints(homology_dimensions_fit)
+
+        if self.n_distances is None:
+            self.n_distances_ = \
+                np.array([X[i].shape[0] for i in range(len(X))]).max()
+        else:
+            self.n_distances_ = self.n_distances
+
+        self._distance_function = DistanceMetric.get_metric(
+            self.metric, **self.metric_params)
+
+        return self
+
+    def _topological_vector(self, Xd):
+        Xv = np.zeros((self.n_distances_,))
+        distances = self._distance_function.pairwise(Xd)
+
+        Xd[:, 1] = Xd[:, 1] - Xd[:, 0]
+        min_persistence = 0.5 * np.minimum(Xd[:, 1], Xd[:, 1].T)
```

Comment:
@gtauzin could you explain this line to me? I expect `Xd[:, 1]` to be a 1d array at this point, and if true this means that `Xd[:, 1]` are `Xd[:, 1].T` are equal arrays.

Reply:
I believe you are right, this line must have remained for my own experiments.

Reply:
Should I change this line to
```
min_persistence = 0.5 * Xd[:, 1]
```
? I also think this logic should be documented.